module2/lessons/js_fundamentals.md

```diff
@@ -133,6 +136,130 @@ let creatureCount = "<p>I have " + quantity + " very fancy" + mythicalCreature +
+Complex: objects and arrays
+
+Technically also:
+- Symbol (new in ECMAScript 6)
+- BigInt (new in ECMAScript 2020)  
+You won't likely need either of these in your time at Turing so we won't dig into them.
+</section>
+
+<section class="dropdown">
+### What do we need to know about each data type?
+
+- Boolean: true or false (not in quotation marks)  
+
+- Undefined: not yet given a value (the value hasn't been defined so its UNdefined)  
+
+- Null: intentional absence of a value  
+
+- Number: regular ol’ numbers as we know them  
+
+- String: any characters wrapped in quotation marks (single or double quotes) - often just normal text/words
+
+- Object: like a hash in Ruby
+
+- Array: like an array in Ruby
+
+</section>
+
+<section class="dropdown">
+### What are the differencs and use cases for undefined vs null?
+Both mean "nothing".  
+Null is like a placeholder where someone proactively decided to specifically declare "nothing".  
+Undefined is kind of unintentional. Like, it's nothing because no one got around to declaring anything so it defaults to "nothing".
+
+A use case for null might be creating a `userEmail` variable and assigning it to null in case the user does not provide an email.   
+
+You aren't likely to specifically assign a value of `undefined` so there's not a great use case to use as an example.  
+</section>
+
+<section class="dropdown">
+### Why do we need variables?
+To be able to store pieces of data in a way that we can reference and use them later.
+</section>
+
+<section class="dropdown">
+### What are best practices for naming variables in JS?
+  - Variable names cannot begin with a number.  They can begin with a letter, dollar sign, or an underscore.  
+  - You cannot use a dash (-) or a period (.) within a variable name
+  - You cannot use keywords or reserved words (such as var or for)
+  - Variables are case sensitive.  
+  - Choose names that describe the kind of information you plan to assign the variable. Call it what it is.
+  - Use camelCase for multiple word names i.e., thisIsMyVariableName
+</section>
+
+<section class="dropdown">
+### What is the difference between concatenation and interpolation?
+Concatenation and interpolation are two different ways to accomplish the same goal. The only different is the syntax - neither is better than the other.
+
+Concatenation:  
+  - Uses the `+` string operator to combine variables and other values (strings, numbers, html tags) to make a big, long string.
+
+```js
+var name = "Hillary";
+var age = 30;
+var location = "Denver";
+
+var concatenatedString = "My friend " + name + "is " + age + "years old and lives in " + location + "."
+```
+
+Interpolation:  
+  - Uses the backticks and `${ }` to inject variables to make a big, long string.  
+
+```js
+var name = "Hillary";
+var age = 30;
+var location = "Denver";
+
+var interpolatedString = `My friend ${name} is ${age} years old and lives in ${Denver}.`
```

Comment:
I am not sure if this is intentional, but inside the interpolation, the location variable has not been used, and “Denver” has been hardcoded.

Reply:
_Thank you!_